### PR TITLE
New Welcome Page

### PR DESCRIPTION
**Pull request by GrabowskiM**

### CODE REVIEW

**Review by tischsoic — APPROVED** [review]

File: ibexa/content/4.4/assets/scss/_main.scss [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                0 22px 24px 0 rgba($ibexa-welcome-color-extra-red, 0.2);
+        }
+    }
+}
+
+.ibexa-welcome-link {
+    text-decoration: underline;
+    color: $ibexa-welcome-color-dark;
+    font-weight: 600;
+    font-size: 14px;
```
Comment (Gengar-i) [inline ibexa/content/4.4/assets/scss/_main.scss:120]: $ibexa-text-font-size-medium ?

File: ibexa/content/4.4/assets/scss/_main.scss [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            box-shadow: 0 0 0 transparent,
+                0 22px 24px 0 rgba($ibexa-welcome-color-extra-red, 0.2);
+        }
+    }
+}
+
+.ibexa-welcome-link {
+    text-decoration: underline;
+    color: $ibexa-welcome-color-dark;
+    font-weight: 600;
```
Comment (Gengar-i) [inline ibexa/content/4.4/assets/scss/_main.scss:119]: $ibexa-font-weight-bold ?

File: ibexa/content/4.4/assets/scss/_main.scss [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+body {
+    font-size: 16px;
+    line-height: 24px;
+    letter-spacing: 0.12px;
+    color: $ibexa-welcome-color-dark;
+    overflow-x: hidden;
+    font-family: 'Noto Sans', sans-serif;
+    font-weight: 400;
```
Comment (Gengar-i) [inline ibexa/content/4.4/assets/scss/_main.scss:13]: $ibexa-font-weight-normal

**Review by dew326 — APPROVED** [review]

File: ibexa/content/4.4/assets/scss/_header.scss [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    .ibexa-welcome-section-header {
+        font-size: 60px;
+        line-height: 82px;
+        margin: 200px 0 40px;
+    }
+
+    .ibexa-welcome-paragraph {
+        font-size: 22px;
+        line-height: 36px;
+        font-weight: 600;
```
Comment (Gengar-i) [inline ibexa/content/4.4/assets/scss/_header.scss:32]: $ibexa-font-weight-bold

File: ibexa/content/4.4/assets/scss/_header.scss [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    .ibexa-welcome-section-header {
+        font-size: 60px;
+        line-height: 82px;
+        margin: 200px 0 40px;
+    }
+
+    .ibexa-welcome-paragraph {
+        font-size: 22px;
```
Comment (Gengar-i) [inline ibexa/content/4.4/assets/scss/_header.scss:30]: $ibexa-text-font-size-extra-large

File: ibexa/content/4.4/assets/scss/_main.scss (hunk 1 above)
Comment (dew326) [reply ibexa/content/4.4/assets/scss/_main.scss:120]: We can't use variables from backoffice on the welcome page.

File: ibexa/content/4.4/assets/scss/_main.scss [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    min-height: 100%;
+}
+
+body {
+    font-size: 16px;
+    line-height: 24px;
+    letter-spacing: 0.12px;
+    color: $ibexa-welcome-color-dark;
+    overflow-x: hidden;
+    font-family: 'Noto Sans', sans-serif;
```
Comment (Gengar-i) [inline ibexa/content/4.4/assets/scss/_main.scss:12]: Not sure if $ibexa-font-family

**Comment by mnocon** [comment]
Will the changes from `webpack.config.js` be overwritten when compatibility-layer is installed? It provides its own version of this file: https://github.com/ibexa/recipes-dev/blob/master/ibexa/compatibility-layer/4.4/encore/webpack.config.js

**Review by Gengar-i — APPROVED** [review]

**Review by lucasOsti — APPROVED** [review]

**Comment by GrabowskiM** [comment]
> Will the changes from `webpack.config.js` be overwritten when compatibility-layer is installed? It provides its own version of this file: https://github.com/ibexa/recipes-dev/blob/master/ibexa/compatibility-layer/4.4/encore/webpack.config.js

Added https://github.com/ibexa/recipes-dev/pull/49/commits/ffbad38e3f3b51687ec40ad4c7ead327bf255d27#diff-fa30278c4dc6bf0ab90743e7fd6cd5a05dd0382e683b8de6f551f4af5b7e5bc7R39

File: ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 7]
```diff
@@ -3,156 +3,307 @@
...
-                        </div>
-                    </div>
+                <h3 class="ibexa-welcome-block-header">
+                    {{ 'ibexa_platform_welcome_page.rest_api.title'|trans|desc('REST API reference') }}
+                </h3>
+                <p class="ibexa-welcome-paragraph">
+                    {{ 'ibexa_platform_welcome_page.rest_api.content'|trans|desc('As an API-first product, Ibexa DXP focuses on well-structured APIs. REST API endpoints cover most functionalities of the product.') }}
+                </p>
+                <a
+                    href="http://doc.ibexa.co/rest-api-reference"
```
Comment (mnocon) [inline ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:213]: This link results in 404 - unless there's a doc change that's required.
Also, shouldn't we use https by default?

**Review by mnocon — CHANGES_REQUESTED** [review]

**Comment by mnocon** [comment]
Nitpick warning:

Zrzut ekranu 2023-01-24 o 14 57 09

To be honest the "Create content" button looks too real, I tried clicking it just to see if it does anything 😄

File: ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 8]
```diff
@@ -3,156 +3,307 @@
...
+
+    <div class="ibexa-welcome-contact">
+        <div class="ibexa-welcome-container">
+            <div class="ibexa-welcome-row ibexa-welcome-contact__block">
+                <div class="ibexa-welcome-column">
+                    <h3 class="ibexa-welcome-block-header">
+                        {{ 'ibexa_platform_welcome_page.contact.title'|trans|desc('Contact to support') }}
+                    </h3>
+                    <p class="ibexa-welcome-paragraph">
+                        {{ 'ibexa_platform_welcome_page.contact.content'|trans|desc('If you need help, contact with our support team') }}
```
Comment (DominikaK) [inline ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:276]:
```suggestion
                        {{ 'ibexa_platform_welcome_page.contact.content'|trans|desc('If you need help, contact our support team') }}
```
The same for other editions.

File: ibexa/content/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 9]
```diff
@@ -3,157 +3,307 @@
...
+                        >
+                    </div>
+                    <h1 class="ibexa-welcome-section-header">
+                        {{ 'ibexa_platform_welcome_page.header.title'|trans({
+                            '%name%': ibexa_render_field(content, 'name'),
+                            '%version%': constant('Ibexa\\Contracts\\Core\\Ibexa::VERSION')
+                        })|desc('Welcome to %name% %version%')|raw }}
+                    </h1>
+                    <p class="ibexa-welcome-paragraph">
+                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('This is the clean installation coming with Ibexa Digital Experience Platform.') }}
```
Comment (DominikaK) [inline ibexa/content/4.4/templates/themes/standard/full/welcome_page.html.twig:35]:
```suggestion
                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('The headless content engine at the core of your digital platform.') }}
```

File: ibexa/experience/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 10]
```diff
@@ -3,157 +3,307 @@
...
+                        >
                     </div>
+                    <h1 class="ibexa-welcome-section-header">
+                        {{ 'ibexa_platform_welcome_page.header.title'|trans({
+                            '%name%': ibexa_render_field(content, 'name'),
+                            '%version%': constant('Ibexa\\Contracts\\Core\\Ibexa::VERSION')
+                        })|desc('Welcome to %name% %version%')|raw }}
+                    </h1>
+                    <p class="ibexa-welcome-paragraph">
+                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('This is the clean installation coming with Ibexa Digital Experience Platform.') }}
```
Comment (DominikaK) [inline ibexa/experience/4.4/templates/themes/standard/full/welcome_page.html.twig:35]:
```suggestion
                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('Engage your customers with memorable experiences.') }}
```

File: ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 11]
```diff
@@ -3,156 +3,307 @@
...
+                        >
                     </div>
+                    <h1 class="ibexa-welcome-section-header">
+                        {{ 'ibexa_platform_welcome_page.header.title'|trans({
+                            '%name%': ibexa_render_field(content, 'name'),
+                            '%version%': constant('Ibexa\\Contracts\\Core\\Ibexa::VERSION')
+                        })|desc('Welcome to %name% %version%')|raw }}
+                    </h1>
+                    <p class="ibexa-welcome-paragraph">
+                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('This is the clean installation coming with Ibexa Digital Experience Platform.') }}
```
Comment (DominikaK) [inline ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:35]:
```suggestion
                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('Digitalize and automate the way you do business.') }}
```

File: ibexa/oss/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 12]
```diff
@@ -3,157 +3,307 @@
...
+                        >
+                    </div>
+                    <h1 class="ibexa-welcome-section-header">
+                        {{ 'ibexa_platform_welcome_page.header.title'|trans({
+                            '%name%': ibexa_render_field(content, 'name'),
+                            '%version%': constant('Ibexa\\Contracts\\Core\\Ibexa::VERSION')
+                        })|desc('Welcome to %name% %version%')|raw }}
+                    </h1>
+                    <p class="ibexa-welcome-paragraph">
+                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('This is the clean installation coming with Ibexa Digital Experience Platform.') }}
```
Comment (DominikaK) [inline ibexa/oss/4.4/templates/themes/standard/full/welcome_page.html.twig:35]:
```suggestion
                        {{ 'ibexa_platform_welcome_page.header.content'|trans|desc('Open-source solution for building custom, scalable websites.') }}
```

File: ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 13]
```diff
@@ -3,156 +3,307 @@
...
+                    <img
+                        src="{{ asset('build/images/user.svg') }}"
+                        alt="{{ 'ibexa_platform_welcome_page.user_documentation.image'|trans|desc('User documentation') }}"
+                    />
+                </div>
+                <h3 class="ibexa-welcome-block-header">
+                    {{ 'ibexa_platform_welcome_page.user_documentation.title'|trans|desc('User documentation') }}
+                </h3>
+                <p class="ibexa-welcome-paragraph">
+                    {{ 'ibexa_platform_welcome_page.user_documentation.content'|trans|desc('Manage content, products, site and content structure and muchh more using the Ibexa DXP administration interface.
```
Comment (mnocon) [inline ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:102]:
```suggestion
                    {{ 'ibexa_platform_welcome_page.user_documentation.content'|trans|desc('Manage content, products, site and content structure and much more using the Ibexa DXP administration interface.
```

Comment (mnocon) [reply ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:102]: (also in other places)

File: ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 14]
```diff
@@ -3,156 +3,307 @@
...
+                    src="{{ asset('build/images/training.svg') }}"
+                    alt="{{ 'ibexa_platform_welcome_page.training_platform.image'|trans|desc('Training platform') }}"
+                />
+            </div>
+            <div class="ibexa-welcome-column">
+                <h3 class="ibexa-welcome-block-header">
+                    {{ 'ibexa_platform_welcome_page.training_platform.title'|trans|desc('Training platform') }}
+                </h3>
+                <p class="ibexa-welcome-paragraph">
+                    {{ 'ibexa_platform_welcome_page.training_platform.content'|trans|desc('Once completed, this wizard will recommend the Ibexa DXP product that best suits your needs and display the base price of the product, including any additional services you have selected
```
[the remark this answers is not in the document]
Comment (juskora) [reply ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:255]: Yes, it's great :)

File: ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig [hunk 15]
```diff
@@ -3,156 +3,307 @@
...
-                        </div>
-                    </div>
+                <h3 class="ibexa-welcome-block-header">
+                    {{ 'ibexa_platform_welcome_page.tutorials.title'|trans|desc('Tutorials') }}
+                </h3>
+                <p class="ibexa-welcome-paragraph">
+                    {{ 'ibexa_platform_welcome_page.tutorials.content'|trans|desc('Practical tutorials lead you through the first project to help you learn product basics.') }}
+                </p>
+                <a
+                    href="https://doc.ibexa.co/en/latest/tutorials/platform_beginner/building_a_bicycle_route_tracker_in_ez_platform"
```
Comment (DominikaK) [inline ibexa/commerce/4.4/templates/themes/standard/full/welcome_page.html.twig:193]:
```suggestion
                    href="https://doc.ibexa.co/en/latest/tutorials/tutorials/"
```

**Review by DominikaK — APPROVED** [review]

**Review by juskora — APPROVED** [review]

**Comment by mnocon** [comment]
The displayed name of the product differs between OSS, Content, Experience and Commerce versions.

OSS, Content:
"Ibexa Platform"
Zrzut ekranu 2023-01-24 o 17 15 06


Experience, Commerce:
Zrzut ekranu 2023-01-24 o 17 15 21

I think it's because of different name for the root Content Item in Content and Experience (in Content it's a folder named "Ibexa Platform", in Experience it's a landing page called "Ibexa Digital Experience Platform").

I think it should be unified across Content, Experience and Commerce editions (or maybe all?) - not sure if it's easy to achieve though (unless we want to hardcode those)

Also, for the doc part, https://doc.ibexa.co/en/latest/getting_started/first_steps/ might need to be updated? The screenshot for sure

**Review by mnocon — APPROVED** [review]

**Comment by webhdx** [comment]
@GrabowskiM have you looked into Sonar Cloud security report? Maybe it's worth fixing.

**Comment by lserwatka** [comment]
> Nitpick warning:
> 
> 
> 
> Zrzut ekranu 2023-01-24 o 14 57 09
> 
> 
> 
> To be honest the "Create content" button looks too real, I tried clicking it just to see if it does anything 😄 
> 
> 
> 
> 

Marek could you pass this to Natalia? We must revisit this.

**Pull request merged**